server/src/main/java/org/opensearch/indices/recovery/PeerRecoveryTargetService.java

```diff
@@ -247,7 +247,14 @@ private void doRecovery(final long recoveryId, final StartRecoveryRequest preExi
                     indexShard.prepareForIndexRecovery();
                     final boolean hasRemoteSegmentStore = indexShard.indexSettings().isRemoteStoreEnabled();
                     if (hasRemoteSegmentStore) {
-                        indexShard.syncSegmentsFromRemoteSegmentStore(false, recoveryTarget::setLastAccessTime);
+                        try {
+                            indexShard.syncSegmentsFromRemoteSegmentStore(false, recoveryTarget::setLastAccessTime);
+                        } catch (Exception e) {
```

Comment:
should we catch `FileNotFoundException` only ? 

Lets also add a ToDo with the github issue linked to remove this later .

Reply:
> should we catch FileNotFoundException only?

Given this is a temporary mitigation, kept it generic. In case there are another exception in segment download logic, we still have a way to fallback to node-to-node segment copy.

> Lets also add a ToDo with the github issue linked to remove this later .

Added a ToDo with GitHub issue link